browser/scripts/ui-core-statestore.js

```diff
@@ -0,0 +1,188 @@
+var uiMode = {
+	build: 	'build',
+	program: 'program'
+}
+var uiModifyMode = {
+	move: 	'translate',
+	rotate: 'rotate',
+	scale: 	'scale'
+}
+
+/**
+ * first-level properties e.g. .mode, .visibility emit events upon changing
+ * second-level properties through .visibility, .panelStates and .context also emit events
+ * if persistentStorageRef, except for a number of ignored properties, the object will persist
+ * @param persistentStorageRef (store or recall state)
+ */
+var UiStateStore = function(persistentStorageRef, context) {
+	EventEmitter.apply(this, arguments)
+	var that = this
+
+	var a = function(o, props, callback) { // ES5
```

Comment:
a() had a very descriptive name, which made things look a bit ridiculous.. i'll tweak it.